File: ios/view/Pages/NewsDetailPage.swift
import SwiftUI

struct NewsDetailPage: View {
    
    @Environment(\.presentationMode) var presentation
    
    private let news: News
    
    init(news: News) {
        self.news = news
    }
    
    var body: some View {
        ZStack(alignment: .leading) {
            MaterialTheme.colors.BACKGROUND.edgesIgnoringSafeArea(.all)
            
            ScrollView() {
                VStack(alignment: .leading) {
                    
                    ZStack(alignment: .topLeading) {
                        
                        AsyncImage(urlString: news.metadata.image)
                            .frame(height: 256)
                            .frame(
                                minWidth: 0,
                                maxWidth: .infinity
                            )
                            .clipShape(Rectangle())
                        
                        VStack {
                            
                            Spacer(minLength: 232)
                                                        
                            Image("background_curve")
                                .frame(
                                    minWidth: 0,
                                    maxWidth: .infinity,
                                    alignment: .bottom
                                )
                                .foregroundColor(MaterialTheme.colors.BACKGROUND)
                        }
                        .frame(height: 256)
                        
                        ZStack {
                            Group {
                                Image("baseline_arrow_back_black_24pt")
                                    .resizable()
                                    .frame(width: 20, height: 20)
                                    .foregroundColor(Color.white)
                            }
                            .frame(width: 32, height: 32, alignment: .center)
                            .background(MaterialTheme.colors.PRIMARY)
                            .cornerRadius(4)
                            .onTapGesture {
                                self.presentation.wrappedValue.dismiss()
                            }
                        }
                        .padding(.all, 16)
                    }
                    
                    Text(news.metadata.title)
                        .modifier(MaterialTheme.typography.h1)
                        .padding(.leading, 24)
                        .padding(.top, 12)
                        .padding(.trailing, 24)
                        .padding(.bottom, 2)
                                        
                    Text(format(from: news.metadata.timestamp))
                        .modifier(MaterialTheme.typography.subtitle1)
                        .padding(.leading, 24)
                        .padding(.top, 2)
                        .padding(.trailing, 24)
                        .padding(.bottom, 24)
                    
                    ForEach(news.content, id: \.self) { element in
                        Group {
                            if element.type == ContentType.HEADER {
                                ContentHeader(header: element.content)
                            } else if element.type == ContentType.TEXT {
                                ContentText(text: element.content)
                            } else if element.type == ContentType.IMAGE {
                                ContentImage(url: element.content)
                            }
                        }
                    }
                    
                    FooterComponent()
                }
            }
        }
        .navigationBarTitle("")
        .navigationBarHidden(true)
    }
    
    func format(from input: Date) -> String {
        let dateTimeFormatter = DateFormatter()
        dateTimeFormatter.dateFormat = "dd. MMMM yyyy"
        return dateTimeFormatter.string(from: input)
    }
}

File: android/view/pages/NewsDetailPage.kt
package com.tobiasheymann.ep.view.pages

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.Icon
import androidx.compose.material.IconButton
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.rounded.ArrowBack
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.ColorFilter
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.unit.dp
import androidx.navigation.NavController
import coil.compose.rememberImagePainter
import coil.size.Scale
import com.tobiasheymann.ep.R
import com.tobiasheymann.ep.model.ContentType
import com.tobiasheymann.ep.model.News
import com.tobiasheymann.ep.view.components.ContentHeader
import com.tobiasheymann.ep.view.components.ContentImage
import com.tobiasheymann.ep.view.components.ContentText
import com.tobiasheymann.ep.view.components.FooterComponent
import java.time.ZoneId
import java.time.format.DateTimeFormatter
import java.util.*

@Composable
fun NewsDetailPage(navController: NavController) {

    if (navController.currentBackStackEntry?.arguments?.containsKey("news") == true) {
        val news = navController.currentBackStackEntry?.arguments?.getSerializable("news") as News

        LazyColumn {
            item {
                Box {
                    Image(
                        painter = rememberImagePainter(
                            data = news.metadata.image,
                            builder = {
                                crossfade(true)
                                scale(Scale.FILL)
                                placeholder(drawableResId = R.drawable.placeholder_square_large)
                            }
                        ),
                        contentDescription = null,
                        contentScale = ContentScale.Crop,
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(256.dp)
                    )
                    Image(
                        painter = painterResource(id = R.drawable.background_curve),
                        contentDescription = null,
                        modifier = Modifier
                            .fillMaxWidth()
                            .align(Alignment.BottomCenter)
                            .offset(y = 5.dp),
                        colorFilter = ColorFilter.tint(MaterialTheme.colors.background)
                    )
                    Box(modifier = Modifier.padding(16.dp)) {
                        IconButton(
                            onClick = {
                                navController.popBackStack()
                            },
                            modifier = Modifier
                                .clip(RoundedCornerShape(4.dp))
                                .background(MaterialTheme.colors.primary)
                                .size(32.dp, 32.dp),
                            content = {
                                Icon(
                                    imageVector = Icons.Rounded.ArrowBack,
                                    contentDescription = null,
                                    modifier = Modifier
                                        .size(20.dp, 20.dp),
                                    tint = Color.White
                                )
                            }
                        )
                    }
                }

                val dateTimeFormatter = DateTimeFormatter
                    .ofPattern("dd. MMMM yyyy")
                    .withLocale(Locale.GERMANY)
                    .withZone(ZoneId.systemDefault())
                Text(
                    text = news.metadata.title,
                    modifier = Modifier.padding(start = 24.dp, top = 12.dp, end = 24.dp, bottom = 2.dp),
                    style = MaterialTheme.typography.h1
                )
                Text(
                    text = dateTimeFormatter.format(news.metadata.timestamp),
                    modifier = Modifier.padding(start = 24.dp, top = 2.dp, end = 24.dp, bottom = 24.dp),
                    style = MaterialTheme.typography.subtitle1
                )

                news.content.forEach { c ->
                    when {
                        c.type === ContentType.HEADER -> {
                            ContentHeader(header = c.content)
                        }
                        c.type === ContentType.TEXT -> {
                            ContentText(text = c.content)
                        }
                        c.type === ContentType.IMAGE -> {
                            ContentImage(url = c.content)
                        }
                    }
                }

                FooterComponent(navController = navController)
            }
        }
    }
}